<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="sicknav350laser">

  <xacro:property name="laser_x" value="0.12" />
  <xacro:property name="laser_y" value="0.12" />
  <xacro:property name="laser_z" value="0.22" />

  <xacro:macro name="nav350laser" params="parent_link lase_link *joint_origin" >
    <!-- Hokuyo Laser -->
    <link name="${lase_link}">
      <collision>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <box size="${laser_x} ${laser_y} ${laser_z}"/>
        </geometry>
      </collision>

      <visual>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
         <mesh filename="package://rrbot_description/meshes/hokuyo.dae"/>
         <!--mesh filename="package://sicknav350/meshes/sicknav350.stl"/-->
        </geometry>
      </visual>

      <inertial>
        <mass value="1e-5" />
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
      </inertial>
    </link>
  
   
  <!-- in real condition .we do not need it -->
  <!-- laser in gazebo -->
    <!--gazebo reference="laser_link">
      <material>Gazebo/blue</material>
      <sensor type="gpu_ray" name="head_laser_sensor">
        <pose>0 0 0 0 0 0</pose>
        <visualize>true</visualize>
        <update_rate>30</update_rate>
        <ray>
          <scan>
            <horizontal>
              <samples>1440</samples>
              <resolution>0.25</resolution>
              <min_angle>-1.570796</min_angle>
              <max_angle>1.570796</max_angle>
            </horizontal>
          </scan>
          <range>
            <min>0.50</min>
            <max>70.0</max>
            <resolution>0.25</resolution>
          </range>
          <noise>
            <type>gaussian</type> 
            <mean>0.0</mean>
            <stddev>0.01</stddev>
          </noise>
        </ray>
        <plugin name="gazebo_ros_head_laser_controller" filename="libgazebo_ros_gpu_laser.so">
          <topicName>/laser/scan</topicName>
          <frameName>laser_scan_frame</frameName>
        </plugin>
      </sensor>
    </gazebo-->
    
    <joint name="lase_joint" type="fixed">
      <parent link="${parent_link}" />
      <child link="${laser_link}" />
      <axis xyz="0 1 0" />
      <!-- origin cord spcified  outsidely -->
      <xacro:insert_block name="joint_origin"/>
    </joint>

  </xacro:macro>

</robot>
